editor/js/CmdAddObject.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @author dforrer / https://github.com/dforrer
+ * Developed as part of a project at University of Applied Sciences and Arts Northwestern Switzerland (www.fhnw.ch)
+ */
+
+/**
+ * @param object THREE.Object3D
+ * @constructor
+ */
+
+CmdAddObject = function ( object ) {
+
+	Cmd.call( this );
+
+	this.type = 'CmdAddObject';
+
+	this.object = object;
+	if ( object !== undefined ) {
+
+		this.name = 'Add Object: ' + object.name;
+
+	}
+
+};
+
+CmdAddObject.prototype = {
+
+	execute: function () {
+
+		this.editor.addObject( this.object );
+		this.editor.select( this.object );
+
+	},
+
+	undo: function () {
+
+		this.editor.removeObject( this.object );
+		this.editor.deselect();
+
+	},
+
+	toJSON: function () {
+
+		var output = Cmd.prototype.toJSON.call( this );
+
+		this.object.updateMatrixWorld( true );
+
```

Comment:
Thanks to your adjustments to Material.toJSON() I was able to simplify CmdSetMaterial.toJSON(). CmdSetMaterial.fromJSON() I have adjusted accordingly. 

See https://github.com/lxxxvi/three.js/commit/257b59650d20b03dbb5902aa4598eac3b9233382

In .fromJSON() we still use the functions ( parseImages, parseTextures, parseMaterials ) provided by THREE.ObjectLoader because THREE.MaterialLoader alone doesn't handle the parsing of textures/images.

**As for other workarounds:** I just looked at the diff again and nothing else stood out. Of course since we wrote these lines of code our judgement certainly cannot be described as impartial. :) 
That's why we welcome your input and the input of the community.